src/reducers/starWarsReducer.js

```diff
@@ -1,13 +1,20 @@
-import /* we need our action types here*/ "../actions";
+import {FETCH, SUCCESS, FAILURE} from "../actions";
 const initialState = {
-  characters: []
+  characters: [], isFetching: false, error: null
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
     // Fill me in with the important reducers
     // action types should be FETCHING, SUCCESS and FAILURE
     // your switch statement should handle all of these cases.
+
+    case FETCH:
+    return {...state, isFetching:true};
+    case SUCCESS:
+    return {...state, isFetching:false, characters: [...state.characters, ...action.payload]}
```

Comment:
```suggestion
    return {...state, isFetching:false, characters: action.payload}
```
We don't need to spread in here because we're always only trying to get an unchanging array from the database. We have no admin control on the database when making an AJAX request to this third-party API.